app/src/main/java/com/github/sdp/mediato/CreateProfileFragment.java

```diff
@@ -0,0 +1,183 @@
+
+        // Generate a username
+        usernameTextInput.setEndIconOnClickListener(generateUsername(usernameTextInput, usernameEditText));
+
+        // Remove the error if the user fix its username
+        usernameEditText.addTextChangedListener(new UsernameWatcher(usernameTextInput, usernameEditText));
+
+        // Create the profile if valid otherwise error
+        createProfileButton.setOnClickListener(tryCreateProfile(usernameTextInput, usernameEditText));
+
+        // Inflate the layout for this fragment
+        return view;
+    }
+
+    @NonNull
+    private View.OnClickListener photoPicker() {
+        return v -> ImagePicker.Companion.with(CreateProfileFragment.this)
+                .crop()
+                .cropSquare()
+                .compress(1024)
+                .maxResultSize(620, 620)
+                .createIntent( intent -> {
+                    photoPickerResult.launch(intent);
+                    return null;
+                });
+    }
+
+    @NonNull
+    private View.OnClickListener generateUsername(TextInputLayout usernameTextInput, TextInputEditText usernameEditText) {
+        return v -> {
+            usernameTextInput.setError(null);
+            Faker faker = new Faker();
+            String animal = faker.animal().name();
+            String number = faker.number().digits(5);
+            usernameEditText.setText(animal.concat(number));
+        };
+    }
+
+    @NonNull
+    private View.OnClickListener tryCreateProfile(TextInputLayout usernameTextInput, TextInputEditText usernameEditText) {
+        return view -> {
+            String errorMsg = getUsernameErrorMsg(usernameEditText.getText());
+            usernameTextInput.setError(errorMsg);
+
+            if (null == errorMsg) {
+                //TODO Add navigation to Add favorite page
+            }
+        };
+    }
+
+    private String getUsernameErrorMsg(@Nullable Editable text) {
+        switch (isUsernameValid(text)) {
+            case NULL:
+                return getString(R.string.mt_username_error_null);
+            case TOO_SHORT:
+                return getString(R.string.mt_username_error_too_short);
+            case TOO_LONG:
+                return getString(R.string.mt_username_error_too_long);
+            case ALREADY_TAKEN:
+                return getString(R.string.mt_username_error_already_taken);
+            default:
+                return null;
+        }
+    }
+
+    private UsernameError isUsernameValid(@Nullable Editable text) {
+        if (text == null) {
+            return UsernameError.NULL;
+        } else if (text.length() < getResources().getInteger(R.integer.mt_username_min_length)) {
+            return UsernameError.TOO_SHORT;
+        } else if (text.length() > getResources().getInteger(R.integer.mt_username_max_length)) {
+            return UsernameError.TOO_LONG;
+        } else {
+            return UsernameError.GOOD;
+        }
+    }
```

Comment:
I think same as the UsernameError enum this should not be defined in this fragment. For example, there is already a method in the Preconditions class that has a method `checkUsername()` that is not implemented yet. There are also already constants defined in a different way than here as an xml. I think for now this is fine but we should discuss how we can handle constants and checking functions in a uniform way in the next meeting.